source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -463,7 +463,7 @@ FilterTrailersStatus Filter::onTrailers(ProcessorState& state, Http::HeaderMap&
     return FilterTrailersStatus::StopIteration;
   }
 
-  if (!body_delivered && state.bodyMode() == ProcessingMode::BUFFERED) {
+  if (!body_delivered && state.bufferedData() && state.bodyMode() == ProcessingMode::BUFFERED) {
```

Comment:
The other utility function hasBufferedData() can not be used here since it not only check bufferedData() not nullptr, also checks the data->length >0.  Using  hasBufferedData() will break the decodeData() with an empty chunk case.

Reply:
QQ: I am wondering when we need to decodeData() with an empty chunk case? What are the use cases for this?

Reply:
bufferedData()!=nullptr